common/djangoapps/embargo/middleware.py

```diff
@@ -73,40 +74,47 @@ def __init__(self):
         # If embargoing is turned off, make this middleware do nothing
         if not settings.FEATURES.get('EMBARGO', False) and not self.site_enabled:
             raise MiddlewareNotUsed()
+        self.enable_country_access = settings.FEATURES.get('ENABLE_COUNTRY_ACCESS', False)
 
     def process_request(self, request):
         """
         Processes embargo requests.
         """
-        url = request.path
-        course_id = course_id_from_url(url)
-        course_is_embargoed = EmbargoedCourse.is_embargoed(course_id)
-
-        # If they're trying to access a course that cares about embargoes
-        if self.site_enabled or course_is_embargoed:
-
-            # Construct the list of functions that check whether the user is embargoed.
-            # We wrap each of these functions in a decorator that logs the reason the user
-            # was blocked.
-            # Each function should return `True` iff the user is blocked by an embargo.
-            check_functions = [
-                self._log_embargo_reason(check_func, course_id, course_is_embargoed)
-                for check_func in [
-                    partial(self._is_embargoed_by_ip, get_ip(request)),
-                    partial(self._is_embargoed_by_profile_country, request.user)
+        if self.enable_country_access:
+            if self.country_access_rules(request):
+                return None
+            else:
+                return self._embargo_redirect_response
+        else:
```

Comment:
just a nitpick but you may be able to avoid so much nesting by removing this 'else', since the if statement above always returns.